crates/attestation/src/crypto/crl.rs

```diff
@@ -0,0 +1,279 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use crate::crypto::SubjectPublicKeyInfoExt;
+
+use anyhow::{bail, Result};
+use der::Encode;
+use x509::crl::CertificateList;
+use x509::PkiPath;
+
+pub trait PkiPathCRLCheck<'a> {
+    fn check_crl(&self, crls: &'a [CertificateList<'a>]) -> Result<()>;
+}
+
+impl<'a> PkiPathCRLCheck<'a> for PkiPath<'a> {
+    fn check_crl(&self, crls: &'a [CertificateList<'a>]) -> Result<()> {
+        let mut iter = self.iter().peekable();
+        loop {
+            let cert = iter.next();
+            let next = iter.peek();
+            if cert.is_none() || next.is_none() {
+                break;
+            }
+            let cert = cert.unwrap();
+            let next = next.unwrap();
+            for crl in crls {
+                // Don't bother with an empty CRL
+                if crl.tbs_cert_list.revoked_certificates.is_none() {
```

Comment:
We need to check this later, after the signature validation: otherwise an attacker could send an invalidly signed empty CRL.
I think this is an optimization that's not going to be that useful in the end.

Reply:
Some of the CRLs are empty when I manually checked. I did this so the subsequent signature checker isn't referencing empty bytes.

Reply:
You mean the entire CRL is empty, or just the list of revoked certificates?

Reply:
So basically: is the CRL file itself empty, or is the part that contains signed certs empty?
Because in the latter case, the signature checker should still be referencing the entire TBSCertList, and all verification should be just fine.

Reply:
Removed.